opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/ResourceRestService.java

```diff
@@ -123,4 +138,21 @@ public ResourceDTO getResourceForNode(@PathParam("nodeCriteria") final String no
 
         return ResourceDTO.fromResource(resource, depth);
     }
+
+
+    @POST
+    @Path("generateId")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response saveResourcesWithId(String[] resources) {
+        if (m_jsonStore == null) {
```

Comment:
Is this condition necessary? If there was no JsonStore to wire in then this bean would not be created by Spring.